### DISCUSSION TITLE
Unable to remove (previous) controlsocket at '/var/run/pdns.controlsocket': Permission denied

### DISCUSSION THREAD

**Discussion by julian70400** [PowerDNS/pdns#13586] [Q&A] [answered]
Since upgrade Debian 10.5 to Debian 11.2, pdns refuse to launch : 

 pdns_server[2295]: Unable to remove (previous) controlsocket at '/var/run/pdns.controlsocket': Permission denied
 systemd[1]: pdns.service: Main process exited, code=exited, status=1/FAILURE

removing /var/run/pdns.controlsocket change nothing..

### Environment
Debian 11.8

Dec 07 15:20:32 PowerDNS Authoritative Server 4.4.1 (C) 2001-2020 PowerDNS.COM BV
Dec 07 15:20:32 Using 64-bits mode. Built using gcc 10.2.1 20210110.
Dec 07 15:20:32 PowerDNS comes with ABSOLUTELY NO WARRANTY. This is free software, and you are welcome to redistribute it according to the terms of the GPL version 2.
Dec 07 15:20:32 Features: libcrypto-ecdsa libcrypto-ed25519 libcrypto-ed448 libcrypto-eddsa libgeoip libmaxminddb lua lua-records PKCS#11 protobuf sodium
Dec 07 15:20:32 Built-in modules:
Dec 07 15:20:32 Configured with: " '--build=x86_64-linux-gnu' '--prefix=/usr' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--sysconfdir=/etc' '--l
ocalstatedir=/var' '--disable-option-checking' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--runstatedir=/run' '--disable-maintainer-mode' '--disable-dependency-tracking' '--sysconfdir=/etc/powerdns' 
'--enable-systemd' '--with-systemd=/lib/systemd/system' '--with-dynmodules=bind ldap lmdb lua2 pipe gmysql godbc gpgsql gsqlite3 geoip remote random tinydns' '--with-modules=' '--enable-ixfrdist' '--
enable-tools' '--with-protobuf' '--enable-unit-tests' '--enable-lua-records' '--enable-experimental-pkcs11' '--enable-reproducible' '--disable-silent-rules' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g 
-O2 -ffile-prefix-map=/build/pdns-s00AVs/pdns-4.4.1=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-z,relro -Wl,-z,now' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS
=-g -O2 -ffile-prefix-map=/build/pdns-s00AVs/pdns-4.4.1=. -fstack-protector-strong -Wformat -Werror=format-security -DPACKAGEVERSION='\''"4.4.1-1.Debian"'\'''"

### Steps to reproduce
Just upgrade your Debian 10.5 to 11.8..

**Comment by Habbie** [comment]
4.4.1 is [not a supported version](https://doc.powerdns.com/authoritative/appendices/EOL.html). Can you try with a newer version? https://repo.powerdns.com/ has packages for you

**Comment by julian70400** [comment]
Ok, 
Same problem with : 
Dec 07 15:54:32 PowerDNS Authoritative Server 4.7.3 (C) 2001-2022 PowerDNS.COM BV
Dec 07 15:54:32 Using 64-bits mode. Built using gcc 12.2.0.
Dec 07 15:54:32 PowerDNS comes with ABSOLUTELY NO WARRANTY. This is free software, and you are welcome to redistribute it according to the terms of the GPL version 2.
Dec 07 15:54:32 Features: libcrypto-ecdsa libcrypto-ed25519 libcrypto-ed448 libcrypto-eddsa libgeoip libmaxminddb lua lua-records PKCS#11 protobuf sodium curl scrypt
Dec 07 15:54:32 Built-in modules:
Dec 07 15:54:32 Configured with: " '--build=x86_64-linux-gnu' '--prefix=/usr' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--sysconfdir=/etc' '--localstatedir=/var' '--disable-option-checking' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--runstatedir=/run' '--disable-maintainer-mode' '--disable-dependency-tracking' '--sysconfdir=/etc/powerdns' '--enable-systemd' '--with-systemd=/lib/systemd/system' '--with-dynmodules=bind ldap lmdb lua2 pipe gmysql godbc gpgsql gsqlite3 geoip remote tinydns' '--with-modules=' '--enable-ixfrdist' '--enable-tools' '--with-protobuf' '--enable-unit-tests' '--enable-lua-records' '--enable-experimental-pkcs11' '--enable-reproducible' '--disable-silent-rules' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -ffile-prefix-map=/build/pdns-gc5TNa/pdns-4.7.3=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-z,relro -Wl,-z,now' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -ffile-prefix-map=/build/pdns-gc5TNa/pdns-4.7.3=. -fstack-protector-strong -Wformat -Werror=format-security -DPACKAGEVERSION='\''"4.7.3-2.Debian"'\'''"

My pdns.conf is : 

allow-axfr-ips= *.*.*.*.*
config-dir=/etc/powerdns
daemon=yes
disable-axfr=no
guardian=yes
local-address=46.105.*.*
local-port=53
master=yes
slave=no
launch=
module-dir=/usr/lib/x86_64-linux-gnu/pdns
setgid=pdns
setuid=pdns
socket-dir=/var/run
version-string=anonymous
include-dir=/etc/powerdns/pdns.d
logging-facility=0
#log-dns-details=yes
#log-dns-queries=yes
loglevel=3
local-ipv6=
gmysql-dnssec=no

**Reply by Habbie** [reply, marked as answer]
> socket-dir=/var/run

please remove this line

**Reply by julian70400** [reply]
Thanks, it works again !

**Comment by julian70400** [comment]
Got another problem, linked to my upgrade, if I am right, the sql schema had been upgraded since my old version : 

 Exiting because communicator thread died with error: virtual void GSQLBackend::getUpdatedMasters(std::vector<DomainInfo>&, std::unordered_set<DNSName>&, CatalogHashMap&) unable to retrieve list of master domains: Could not prepare statement: select d.id, d.name, d.type, d.notified_serial,d.options, d.catalog,r.content from records r join domains d on r.domain_id=d.id and r.name=d.name where r.type='SOA' and r.disabled=0 and d.type in ('MASTER', 'PRODUCER'): Unknown column 'd.options' in 'field list'
 
 What should I do, now ? Where is the new scheme ?

MariaDB [pdns]> DESC domains;
+-----------------+------------------+------+-----+---------+----------------+
| Field           | Type             | Null | Key | Default | Extra          |
+-----------------+------------------+------+-----+---------+----------------+
| id              | int(11)          | NO   | PRI | NULL    | auto_increment |
| name            | varchar(255)     | NO   |     | NULL    |                |
| master          | varchar(128)     | YES  |     | NULL    |                |
| last_check      | int(11)          | YES  |     | NULL    |                |
| type            | varchar(6)       | NO   |     | NULL    |                |
| notified_serial | int(10) unsigned | YES  |     | NULL    |                |
| account         | varchar(40)      | YES  |     | NULL    |                |
+-----------------+------------------+------+-----+---------+----------------+


THanks for your help, it's my master (got a slave with 4.1 currently running), but if I can start it, I will have to rollback :(

**Reply by Habbie** [reply]
please see https://doc.powerdns.com/authoritative/upgrading.html

**Reply by julian70400** [reply]
omg.. Thanks ! sources save also me : 
![image](https://github.com/PowerDNS/pdns/assets/43498802/8382b9f2-6b90-4578-b350-19171856fae1)
